includes/class-alma-wc-generic-handler.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Alma generic handler
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+/**
+ * Alma_WC_Generic_Handler
+ */
+class Alma_WC_Generic_Handler {
+	/**
+	 * Logger
+	 *
+	 * @var Alma_WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * Min amount.
+	 *
+	 * @var float
+	 */
+	private $min_amount;
+
+	/**
+	 * Max amount.
+	 *
+	 * @var float
+	 */
+	private $max_amount;
+
+	/**
+	 * __construct
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+		$this->logger = new Alma_WC_Logger();
+
+		$this->min_amount = alma_wc_get_min_eligible_amount_according_to_settings();
+		$this->max_amount = alma_wc_get_max_eligible_amount_according_to_settings();
+	}
+
+
+	/**
+	 * Check whether we're in a situation where we can inject JS for our payment plan widget
+	 *
+	 * @return bool
+	 */
+	private function is_usable() {
+		if ( ! alma_wc_plugin()->settings->is_usable() ) {
```

Comment:
On peut se fendre d'un `$this->settings = alma_wc_plugin()->settings` dans le constructeur pour cleaner un peu le code de cette classe, qui accède beaucoup aux settings ^^